tests/integration/java-integration/src/test/java/io/fluentbit/test/LoggingDestinationServer.java

```diff
@@ -0,0 +1,116 @@
+/*  Fluent Bit
+ *  ==========
+ *  Copyright (C) 2019-2021 The Fluent Bit Authors
+ *  Copyright (C) 2015-2018 Treasure Data Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package io.fluentbit.test;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.UncheckedIOException;
+import java.net.InetAddress;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.concurrent.TimeUnit;
+
+import static io.fluentbit.test.Util.waitFor;
+import static org.junit.Assert.assertFalse;
+
+public class LoggingDestinationServer implements AutoCloseable {
+    private final ExpectedLogData expectedLogData;
+    private final Server server;
+
+    private volatile boolean hasErrors = false;
+
+    public LoggingDestinationServer(ExpectedLogData expectedLogData) {
+        this.expectedLogData = expectedLogData;
+        server = new Server();
+        new Thread(server, "logging-dest-server-main").start();
+        waitFor(TimeUnit.SECONDS, 5, () -> server.running);
```

Comment:
Does this always wait or only wait up to 5 seconds? I'm not a fan of picking numbers that work locally but do not for others.

Reply:
Again, up to 5 seconds